pkg/controller/deployment/deployment_controller.go

```diff
@@ -493,7 +493,8 @@ func (dc *DeploymentController) updateDeploymentAndClearRollbackTo(deployment *e
 }
 
 func (dc *DeploymentController) syncRecreateDeployment(deployment extensions.Deployment) error {
-	newRS, oldRSs, err := dc.getAllReplicaSets(deployment)
+	// Don't create a new RS if not already existed, so that we avoid scaling up before scaling down
+	newRS, oldRSs, err := dc.getAllReplicaSets(deployment, false)
```

Comment:
Yes it'd be simpler if we set recreate deployment's RS replicas to 0 when creating it https://github.com/kubernetes/kubernetes/pull/20696/files#diff-a5ab8c704a7cc7eb6ea0616cc18829c5R268, which is how I implemented it earlier. 

But aren't we trying to assign correct number of replicas to new RSs at creation time?